ERCS/erc-7629.md

```diff
@@ -0,0 +1,296 @@
+---
+eip: 7629
+title: Unified Token
+description: Unifying ERC-721 and ERC-20 for seamless asset conversion.
+author: 0xZeus1111 (@0xZeus1111), Nvuwa (@Nvuwa)
+discussions-to: https://ethereum-magicians.org/t/erc-7629-unified-token-protocol/18793
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-02-18
+requires: 20, 721
+---
+
+
+## Abstract
+
+[ERC-7629](./eip-7629.md), known as the Unified Token, introduces a comprehensive protocol unifying the characteristics of [ERC-721](./eip-721.md) and [ERC-20](./eip-20.md) tokens within the Ethereum ecosystem. This standard seamlessly integrates the liquidity features of [ERC-20](./eip-20.md) with the non-fungible nature of [ERC-721](./eip-721.md), enabling frictionless conversion between these asset types. [ERC-7629](./eip-7629.md) offers a multifunctional solution, providing developers and users with the flexibility to leverage both liquidity and non-fungibility in a unified token framework.
+
+
+## Motivation
+
+The motivation behind [ERC-7629](./eip-7629.md) stems from the inherent need within the blockchain community for assets that possess both the liquidity of [ERC-20](./eip-20.md) tokens and the non-fungibility of [ERC-721](./eip-721.md) tokens. Current standards present a dichotomy, necessitating users to choose between these features. [ERC-7629](./eip-7629.md) addresses this limitation by providing a unified token standard, empowering users to seamlessly transition between [ERC-20](./eip-20.md) and [ERC-721](./eip-721.md) characteristics, catering to diverse blockchain applications.
+
+## Specification
+
+- [ERC-7629](./eip-7629.md) standardizes a token contract that encompasses features from both [ERC-20](./eip-20.md) and [ERC-721](./eip-721.md).
+- The token contract supports state transitions between [ERC-20](./eip-20.md) and [ERC-721](./eip-721.md) modes, ensuring smooth conversion and utilization of both liquidity and non-fungibility.
+- Defines essential functions and events to support token interactions, conversions, and queries.
+- Implements a low gas consumption [ERC-20](./eip-20.md) mode to maintain gas efficiency comparable to typical [ERC-20](./eip-20.md) token transfers.
+
+Compliant contracts MUST implement the following Solidity interface:
+
+```solidity
+
+pragma solidity ^0.8.0;
+/**
+ * @title ERC-7629 Unify Token Interface
+ * @dev This interface defines the ERC-7629 Unify Token, which unifies ERC-721 and ERC-20 assets.
+ */
+interface IERC7629 {
+    // ERC-20 Transfer event
+    event ERC20Transfer(
```

Comment:
```suggestion
    event Transfer(
```
For full ERC20 backward compatibility

Reply:
The Transfer event for ERC-20 and ERC-721 is the same. I'd like to retain ERC20Transfer while also adding the Transfer event definition.

Reply:
I would like to have a clarification on whether this ERC proposal aims to ensure **complete backward compatibility** with ERC20.